test-requirements.txt

```diff
@@ -1,4 +1,4 @@
-pytest >= 3.3  # for catchlog fixture
+pytest == 3.7.4  # for catchlog fixture
```

Comment:
Is there a particular reason for pinning to *exactly* 3.7.4 rather than just using `>=`?

Reply:
This was just a workaround to see the real errors. See https://github.com/python-trio/trio/pull/650
The change has been reverted and is done in PR#650

Reply:
Thanks! When making a change such as pinning to a particular version it's useful to users to make a comment (in the commit message) as to why so they know whether or not they can safely ignore it e.g. because the underlying issue has been fixed.

Fixing it in #650 is great because now the blame https://github.com/jmfrank63/trio/blame/528a2986434d644be8138219f0f3087e111d79e8/test-requirements.txt#L1 is linked to the PR and so a user can follow that all the way to #651 to see if the problem has been fixed and they're safe to use pytest 3.8.1 with trio.